PowerShell/JumpCloud Commands Gallery/Windows Commands/Windows - Install CrowdStrike Falcon Agent.md

```diff
@@ -0,0 +1,60 @@
+#### Name
+
+Windows - Install CrowdStrike Falcon Agent | v1.0 JCCG
+
+#### commandType
+
+windows
+
+#### Command
+
+```
+$CID="ENTER CID HERE"
+$installerURL="ENTER URL TO FALCON AGENT INSTALLER HERE"
+
+##### Do Not Edit Below This Line #####
+
+$installerTempLocation="C:\Windows\Temp\CSFalconAgentInstaller.exe"
+
+if(Get-Service "CSFalconService" -ErrorAction SilentlyContinue) {
+    Write-Host("Falcon Agent already installed, nothing to do.")
+    exit 0
+}
+Write-Host("Falcon Agent not installed.")
+
+Write-Host("Downloading Falcon Agent installer now.")
+try {
+    Invoke-WebRequest -Uri $installerURL -OutFile $installerTempLocation
+}
+catch {
+    Write-Error("Unable to download Falcon Agent installer.")
+    exit 1
+}
+Write-Host("Finished downloading Falcon Agent installer.")
+
+Write-Host("Installing Falcon Agent now, this may take a few minutes.")
+try {
+    $args = @("/install","/quiet","/norestart","CID=$CID") 
+    $installerProcess = Start-Process -FilePath $installerTempLocation -Wait -PassThru -ArgumentList $args
+}
+catch {
+    Write-Error("Failed to run Falcon Agent installer.")
+    exit 1
+}
+Write-Host("Falcon Agent installer returned $($installerProcess.ExitCode).")
+
+exit $installerProcess.ExitCode
+
+```
+
+#### Description
+
+This command will download and install the CrowdStrike Falcon Agent to the device if it isn't already installed.  In order to use this command, you must know your Customer ID (CID) and have a valid download URL for the Falcon Agent Installer.
```

Comment:
We should note that the Customer ID Checksum is required here, just the Customer ID is not sufficient. I found the following error on the install log when I entered *just* the customer ID:
`10:44:53]i000: Missing or invalid customer ID and/or checksum. Please provide 'CID=MyCIDWithChecksumValue' on the command line for silent installs.`

![Screen Shot 2022-01-28 at 10 53 09 AM](https://user-images.githubusercontent.com/54448601/151597173-775d8e04-58b0-4584-81c0-bd339d9b6853.png)

Lastly, commands will error out in 124 on the command result window with a timeout if you do anything on a slow network, install most applications or any computationally intensive task. This is an annoyance for customers because they think error 124 is a failure but the command will continue despite running past the timeout window. 

We should state something like: "This command may timeout with a 124 error code in the command result window if not extended to account for network or system variability. Extend the command timeout to some value that makes sense in your environment"

Reply:
Hmmm, I must have tested it myself with the CID+Checksum.  I'll be sure to note that.
I will update the description with a note about increasing the timeout.
Thanks!